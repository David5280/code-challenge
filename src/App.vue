<template>
  <div id="app" :class='this.theme'>
    <div class='title'>
      <h1>Tip Calculator</h1>
    </div>
    <div class='container'>
      <Calculator @calculate:tip='calculateTip'/>
      <ResultsContainer :results='results'/>
    </div>
  </div>
</template>

<script>
import Calculator from './components/Calculator.vue';
import ResultsContainer from './components/ResultsContainer.vue';

export default {
  name: 'app',
  methods: {
    calculateTip(billInfo) {
      const tipAmount = billInfo.totalBill * billInfo.tipPercentage;
      const finalBillAmount = 
        parseInt(billInfo.totalBill) + 
        parseInt(tipAmount);
        if (billInfo.partySize) {
        const pricePerPerson = finalBillAmount / billInfo.partySize;
        this.results.pricePerPerson = pricePerPerson.toFixed(2);
        }
      this.results.tipAmount = tipAmount.toFixed(2);
      this.results.finalBillAmount = finalBillAmount.toFixed(2);
      this.determineTheme(billInfo.tipPercentage);
    },
    determineTheme(tipPercentage) {
      if (tipPercentage <= .15) {
        this.theme = 'theme1'
      } else if (tipPercentage > .15 && tipPercentage <= .30) {
        this.theme = 'theme2'
      } else if (tipPercentage > .30) {
        this.theme = 'theme3'
      }
    }
  },
  data() {
    return {
      results: {
        tipAmount: '',
        finalBillAmount: '',
        pricePerPerson: '',
      },
      theme: ''
    }
  },
  components: {
    Calculator,
    ResultsContainer
  }
};
</script>

<style>
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  height: 100vh;
  width: 100vw;
  display: flex;
  flex-direction: column;
  background-color: rgb(211, 211, 211);
  align-items: center;
}
.title {
  background-color: rgba(211, 211, 211, 0.8);
  margin: 50px 0px;
  height: fit-content;
}

.container {
  background-color: rgba(240, 236, 236, 0.8);
  min-width: 300px;
  max-width: 500px;
  width: 25vw;
  height: 375px;
  box-shadow: 1px 1px 15px black;
  border-radius: 3px;
  border: 1px solid black;
  padding: 20px;
}

.theme1 {
  background-color: red;
  background-image: url('./assets/images/no-money.png');
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
}

.theme2 {
  background-image: url('./assets/images/good-tip.png');
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
}

.theme3 {
  background-image: url('./assets/images/big-baller.jpg');
  background-size: cover;
  background-position: center;
}


</style>
