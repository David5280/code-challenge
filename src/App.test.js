import { mount } from '@vue/test-utils';
import App from './App.vue';

describe('App', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(App);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  test('should match the snapshot', () => {
    const wrapper = mount(App);
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
  test('calculateTip accurately calculates tip', () => {
    const wrapper = mount(App);
    const billInfo = {
      totalBill: '1000',
      tipPercentage: '.20',
      partySize: '',
    };
    wrapper.vm.calculateTip(billInfo);
    expect(wrapper.vm.results.tipAmount).toEqual('200.00');
  });
  test('calculateTip accurately calculates finalBill', () => {
    const wrapper = mount(App);
    const billInfo = {
      totalBill: '1000',
      tipPercentage: '.20',
      partySize: '',
    };
    wrapper.vm.calculateTip(billInfo);
    expect(wrapper.vm.results.finalBillAmount).toEqual('1200.00');
  });
  test('calculateTip accurately calculates pricePerPerson', () => {
    const wrapper = mount(App);
    const billInfo = {
      totalBill: '1000',
      tipPercentage: '.20',
      partySize: '2',
    };
    wrapper.vm.calculateTip(billInfo);
    expect(wrapper.vm.results.pricePerPerson).toEqual('600.00');
  });
  test('determineTheme produces the correct theme based on tip amount', () => {
    const wrapper = mount(App);
    const billInfo = {
      totalBill: '1000',
      tipPercentage: '.40',
      partySize: '2',
    };
    wrapper.vm.determineTheme(billInfo.tipPercentage);
    expect(wrapper.vm.theme).toEqual('theme3');
  });
});
