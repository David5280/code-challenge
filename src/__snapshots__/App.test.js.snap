// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  class=""
  id="app"
>
  <h1>
    Tip Calculator
  </h1>
   
  <div
    class="container"
  >
    <div
      class="calculator"
    >
      <form>
        <input
          class="calculator-input"
          id="bill-amount-input"
          name="card"
          placeholder="Bill Amount"
          type="text"
          value=""
        />
         
        <select
          class="calculator-input"
          id="tip-amount-input"
          name="Tip Percentage"
          placeholder="Tip Percentage"
        >
          <option
            disabled="disabled"
            selected="selected"
            value=""
          >
            Tip Percentage
          </option>
           
          <option
            value=".05"
          >
            5%
          </option>
           
          <option
            value=".10"
          >
            10%
          </option>
           
          <option
            value=".15"
          >
            15%
          </option>
           
          <option
            value=".20"
          >
            20%
          </option>
           
          <option
            value=".25"
          >
            25%
          </option>
           
          <option
            value=".30"
          >
            30%
          </option>
           
          <option
            value=".35"
          >
            35%
          </option>
           
          <option
            value=".40"
          >
            40%
          </option>
           
          <option
            value=".45"
          >
            45%
          </option>
           
          <option
            value=".50"
          >
            50%
          </option>
        </select>
         
        <input
          class="calculator-input"
          placeholder="Party Size (Optional)"
          type="text"
          value=""
        />
         
        <div
          class="error-container"
        >
          <!---->
        </div>
         
        <button
          class="calculator-submit"
        >
          Calculate
        </button>
      </form>
    </div>
     
    <div
      class="results-container"
    >
      <div
        class="result-container"
      >
        <p>
          Tip:
        </p>
         
        <!---->
      </div>
       
      <div
        class="result-container"
      >
        <p>
          Total:
        </p>
         
        <!---->
      </div>
       
      <!---->
    </div>
  </div>
</div>
`;
